bundles/org.openhab.binding.gpio/src/main/java/org/openhab/binding/gpio/handler/GPIODigitalInputHandler.java

```diff
@@ -0,0 +1,189 @@
+
+    /** The logger. */
+    private Logger logger = LoggerFactory.getLogger(GPIODigitalInputHandler.class);
+
+    /** The JPigpio instance. */
+    private JPigpio jPigpio;
+
+    /** The gpio pin handled by this handler. */
+    private GPIO gpio;
+
+    /** The last change. */
+    private Date lastChanged;
+
+    /**
+     * Instantiates a new GPIO digital input handler.
+     *
+     * @param thing the thing
+     */
+    public GPIODigitalInputHandler(Thing thing) {
+        super(thing);
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see
+     * org.eclipse.smarthome.core.thing.binding.ThingHandler#handleCommand(org.eclipse.smarthome.core.thing.ChannelUID,
+     * org.eclipse.smarthome.core.types.Command)
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            Channel channel = getThing().getChannel(THING_TYPE_DIGITAL_INPUT_CHANNEL);
+            if (channel == null) {
+                logger.error("Cannot find Channel: " + THING_TYPE_DIGITAL_INPUT_CHANNEL);
+                return;
+            }
+            updateState(channel.getUID(), getValue());
+        } catch (PigpioException e) {
+            logger.error("An error occured while changing the gpio value", e);
+        }
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see org.eclipse.smarthome.core.thing.binding.BaseThingHandler#initialize()
+     */
+    @Override
+    public void initialize() {
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE);
+            logger.error("Bridge is null");
+            return;
+        }
+        PigpioBridgeHandler handler = (PigpioBridgeHandler) bridge.getHandler();
+        if (handler == null) {
+            updateStatus(ThingStatus.OFFLINE);
+            logger.error("Handler is null");
+            return;
+        }
+        jPigpio = handler.getJPiGpio();
+        try {
+            gpio = new GPIO(jPigpio, Integer.parseInt(getConfig().get(GPIO_ID).toString()), 1);
+            jPigpio.gpioSetAlertFunc(gpio.getPin(), new Alert() {
+
+                @Override
+                public void alert(int gpio, int level, long tick) {
+                    lastChanged = new Date();
+
+                    Date thisChange = new Date();
+                    OnOffType state;
+                    try {
+                        state = getValue();
+                    } catch (PigpioException e1) {
+                        logger.error("Unknown pigpio exception", e1);
+                        return;
+                    }
+                    ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+
+                    Runnable runnable = new Runnable() {
+
+                        @Override
+                        public void run() {
+                            try {
+                                if ((boolean) getConfig().get(STRICT_DEBOUNCING)) {
```

Comment:
Don't call `getConfig()` everywhere. The config doesn't change after initialize so you can store the config in the initialize by creating a configuration class and use `getConfigAs`.